### PR TITLE
QON startup time improvements and general perf improvements

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

File: src/helpers/qownnotesmarkdownhighlighter.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ class QOwnNotesMarkdownHighlighter : public MarkdownHighlighter {
    private:
     Sonnet::WordTokenizer *wordTokenizer;
     Sonnet::LanguageFilter *languageFilter;
-    QOwnSpellChecker *spellchecker;
+    QOwnSpellChecker* spellchecker;
```
Comment (pbek) [inline src/helpers/qownnotesmarkdownhighlighter.h]: clang-formatter will not like that :laughing:

Comment (Waqar144) [reply src/helpers/qownnotesmarkdownhighlighter.h]: 😅

Comment (pbek) [reply src/helpers/qownnotesmarkdownhighlighter.h]: `./scripts/clang-format-project.sh` is your friend :wink:

Comment (Waqar144) [reply src/helpers/qownnotesmarkdownhighlighter.h]: It will take a while getting used to doing the extra step before the commit

**Comment by pbek** [comment]
It isn't that important, it only was funny because you were the one who
proposed the formatter. 😁

Waqar Ahmed <notifications@github.com> schrieb am Do., 13. Feb. 2020, 12:41:

**Pull request merged**